api/server/error_response.go

```diff
@@ -63,9 +68,14 @@ func writeV1Error(ctx context.Context, w http.ResponseWriter, statuscode int, er
 	}
 }
 
-// handleV1ErrorResponse used to handle response errors in the same way.
+// HandleErrorResponse used to handle response errors in the same way.
 func HandleErrorResponse(ctx context.Context, w http.ResponseWriter, err error) {
 	log := common.Logger(ctx)
+	if ctx.Err() == context.Canceled {
+		log.Info("request canceled")
+		return
+	}
```

Comment:
bit nitty: for either of the error response handlers, might we move this down to just preceding `WriteError`, which should also log the error? this keeps control flow simpler for logging when we could get a canceled context + some other error that may explain slightly better how the cancellation happened, and no early return to worry about. i.e. last statement for this function:

```go
if ctx.Err() != context.Canceled {
  WriteError(ctx, w, statusCode, err)
}
```

I guess this means we were getting an EOF from writing the response back? it seems like this would also induce an EOF on the client end itself if we did get a cancellation for a reason other than the client hanging up? I'm a bit curious about the root cause of the EOF anyway, I'm not sure us skipping writing back a 500 to the client is going to resolve the EOF?

Reply:
(with a comment about this, preferably)

Reply:
I don't see another reason that we would have a request context being canceled other than the client aborted the request. This fix is not about fixing EOFs on the client side (those are most likely caused by proxies or LBs timing out in front of the server and beyond the current scope). The intention with the fix is to stop us from logging errors/pushing http code stats for client disconnects as if we had an internal server error, which in this case it wasn't.

Reply:
I agree on the nit. Will fix it tomorrow.

Reply:
+1 on adding stats for disconnects as well. This is going to be the first step, just fixing our logs. More PRs will follow to extend our stats :)

Reply:
Actually, going over the nit, I realized that it won't work. I want to bypass the line 
```
		log.WithError(err).WithFields(logrus.Fields{"stack": string(debug.Stack())}).Error("internal server error")
```
to avoid logging client disconnects as internal server errors. So I think early return is necessary I'm afraid. Unless you have other suggestions?

Reply:
ah, gotcha. i was advocating for that i guess. not sure i understand, i guess the error will show 'canceled' but the log message is still internal server error which is misleading since the client hung up and it's not the server's fault? this just slightly prunes the logs to make it not look like the server is on fire so that when 'internal server error' is getting rattled off we know it's actually something? i buy that. carry on

Reply:
Yep, that's it :)